db/migrate/20160318213755_create_gem_download.rb

```diff
@@ -0,0 +1,10 @@
+class CreateGemDownload < ActiveRecord::Migration
+  def change
+    create_table :gem_downloads do |t|
+      t.integer :rubygem_id, null: false
+      t.integer :version_id, null: false
```

Comment:
Yeah, I didnt like much the `0` values. But I couldnt come up with a better solution.
The polymorphic assoc is interesting, TBH, I am not a big fan of polymorphic assocs, but I might give it a shot. Thanks